servers/publikator/lib/mailchimp/updateCampaign.js

```diff
@@ -0,0 +1,54 @@
+const fetch = require('isomorphic-unfetch')
+const debug = require('debug')('publikator:lib:mailchimp:updateCampaign')
+
+const {
+  MAILCHIMP_URL,
+  MAILCHIMP_API_KEY,
+  MAILCHIMP_CAMPAIGN_CONFIGS,
+  DEFAULT_MAIL_FROM_NAME,
+  DEFAULT_MAIL_FROM_ADDRESS
+} = process.env
+
+const mailchimpCampaignConfigs = MAILCHIMP_CAMPAIGN_CONFIGS
+  ? JSON.parse(MAILCHIMP_CAMPAIGN_CONFIGS)
+  : []
+
+module.exports = async ({ campaignId, campaignConfig = {} }) => {
+  const config = {
+    ...mailchimpCampaignConfigs.find(c => c.key === campaignConfig.key),
+    ...campaignConfig
+  }
+
+  const body = {
+    recipients: {
+      ...config.list_id && { list_id: config.list_id },
+      segment_opts: {
+        ...config.saved_segment_id && { saved_segment_id: Number(config.saved_segment_id) }
+      }
+    },
```

Comment:
What if the config doesn't specify `list_id` or `saved_segment_id`? Does sending mailchimp an empty `recipients` key reset what was maybe set manually on mailchimp?

Reply:
Good question, really. But no, it will not overwrite anything if either prop is not posted.

(If `list_id` and `saved_segment_id` is set, this _will_ overwrite whatever one might have changed on MailChimp (same with content). This is intended, declaring Publikator as the single source of truth here.)